### PR TITLE
Backport changes related to CI migration to GitHub Actions to 1.12.x

### PR DESCRIPTION
**Pull request by kubawi**

### CODE REVIEW

File: .github/workflows/build-vault-oss.yml [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+      vault-version:
+        type: string
+        required: true
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    name: Vault ${{ inputs.goos }} ${{ inputs.goarch }} v${{ inputs.vault-version }}
+    steps:
+      - uses: actions/checkout@v3
```
Comment (modrake) [inline .github/workflows/build-vault-oss.yml:43]: Per security request, all actions need to be pinned to a specific sha (.github/workflows/actionlint.yml in this PR has done this correctly)

File: .github/workflows/enos-run-k8s.yml [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+jobs:
+  enos:
+    name: Integration
+    runs-on: ubuntu-latest
+    env:
+      GITHUB_TOKEN: ${{ secrets.ELEVATED_GITHUB_TOKEN }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
```
Comment (modrake) [inline .github/workflows/enos-run-k8s.yml:34]: another case of action sha pinning missing

File: .github/workflows/test-enos-scenario-ui.yml [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+  get-metadata:
+    name: Get metadata
+    runs-on: ubuntu-latest
+    outputs:
+      go-version: ${{ steps.get-metadata.outputs.go-version }}
+      node-version: ${{ steps.get-metadata.outputs.node-version }}
+      runs-on: ${{ steps.get-metadata.outputs.runs-on }}
+      vault_edition: ${{ steps.get-metadata.outputs.vault_edition }}
+    steps:
+      - uses: actions/checkout@v3
```
Comment (modrake) [inline .github/workflows/test-enos-scenario-ui.yml:40]: sha missing

File: .github/workflows/test-run-enos-scenario-matrix.yml [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+      ENOS_VAR_artifactory_username: ${{ secrets.ARTIFACTORY_USER }}
+      ENOS_VAR_artifactory_token: ${{ secrets.ARTIFACTORY_TOKEN }}
+      ENOS_VAR_terraform_plugin_cache_dir: ./support/terraform-plugin-cache
+      ENOS_VAR_vault_build_date: ${{ needs.metadata.outputs.build-date }}
+      ENOS_VAR_vault_product_version: ${{ needs.metadata.outputs.version }}
+      ENOS_VAR_vault_revision: ${{ inputs.vault-revision }}
+      ENOS_VAR_vault_bundle_path: ./support/downloads/${{ inputs.build-artifact-name }}
+      ENOS_VAR_vault_license_path: ./support/vault.hclic
+    steps:
+      - uses: actions/checkout@v3
```
Comment (modrake) [inline .github/workflows/test-run-enos-scenario-matrix.yml:108]: sha

**Review by modrake — APPROVED** [review]
Found a handful of github actions that weren't sha pinned per the security team's req - noted in each workflow where I found at least one but didn't call out each occurrence to avoid an excess of chatter here.

File: .github/workflows/build-vault-oss.yml (hunk 1 above)
Comment (kubawi) [reply .github/workflows/build-vault-oss.yml:43]: Hi, thanks for the review @modrake! I've raised this PR accidentally earlier today and I just realised that. I will close this one, but will make sure to address your comments on the correct one (19770).

**Comment by kubawi** [comment]
This PR was raised accidentally (I will blame GH running slowly today 😇). The correct PR is 19770.

**Pull request closed without merging**